### PR TITLE
Allow mqtt update entity to report progress

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Allow an mqtt update entity to report progress via a JSON payload.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/129468
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Expanded documentation for the MQTT Update platform, including new JSON payload fields: `in_progress`, `update_percentage`, and `display_precision`.
	- Updated examples demonstrating the use of the new fields for better clarity.
	- Enhanced MQTT integration documentation with a new abbreviation for `display_precision` and updated requirements for device configuration sharing.
	- Improved naming conventions for unnamed MQTT entities to derive meaningful names from their device class.

- **Bug Fixes**
	- Minor corrections made throughout the documentation to clarify the functionality of the JSON payloads and improve user understanding.

### CODE REVIEW

File: source/_integrations/update.mqtt.markdown [hunk 1]
```diff
@@ -252,7 +257,99 @@ JSON can also be used as `state_topic` payload.
...
+  description: URL pointing to an image of the update to be applied as entity picture.
+  required: false
+  type: string
+in_progress:
+  description: Boolean to report an update is in progress or not.
+  required: false
+  default: false
+  type: boolean
+update_percentage:
+  description: Number between 0 and 100 to report the update process. A `null` value resets the in progress state.
```
Comment (c0ffeeca7) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
  description: Number between 0 and 100 to report the update process. A `null` value resets the in-progress state.
```
What do you mean, it reset's the in-progress state? it hasn't started? Or it is restarting?

Comment (c0ffeeca7) [reply source/_integrations/update.mqtt.markdown]: ah. null, not zero.

**Review by c0ffeeca7 — DISMISSED** [review]
Thank you, @jbouwh :+1:

**Review by c0ffeeca7 — APPROVED** [review]

**Comment by jbouwh** [comment]
Thanks!

**Pull request merged**